test/suite/test_cursor_bound14.py

```diff
@@ -0,0 +1,176 @@
+            self.set_bounds(cursor, 50, "upper", self.upper_inclusive)
+
+        # Test bound API: test inserting key outside of bounds.
+        cursor.set_key(self.gen_key(15))
+        cursor.set_value(self.gen_val(120))
+        if (self.lower_bounds):
+            self.assertRaisesHavingMessage(
+                wiredtiger.WiredTigerError, lambda: cursor.insert(), '/item not found/')
+        else:
+            self.assertEqual(cursor.insert(), 0)
+
+        cursor.set_key(self.gen_key(90))
+        cursor.set_value(self.gen_val(120))
+        if (self.upper_bounds):
+            self.assertRaisesHavingMessage(
+                wiredtiger.WiredTigerError, lambda: cursor.insert(), '/item not found/')
+        else:
+            self.assertEqual(cursor.insert(), 0)
+
+        # Test bound API: test updating an existing key outside of bounds.
+        cursor.set_key(self.gen_key(10))
+        cursor.set_value(self.gen_val(120))
+        if (self.lower_bounds):
+            self.assertEqual(cursor.update(), wiredtiger.WT_NOTFOUND)
+        else:
+            self.assertEqual(cursor.update(), 0)
+        
+
+        cursor.set_key(self.gen_key(95))
+        cursor.set_value(self.gen_val(120))
+        if (self.upper_bounds):
+            self.assertEqual(cursor.update(), wiredtiger.WT_NOTFOUND)
+        else:
+            self.assertEqual(cursor.update(), 0)
+
+        # Test bound API: test reserve on an existing key with bounds.
+        self.session.begin_transaction()
+        cursor.set_key(self.gen_key(10))
+        if (self.lower_bounds):
+            self.assertRaisesHavingMessage(
+                wiredtiger.WiredTigerError, lambda: cursor.reserve(), '/item not found/')
+        else:
+            self.assertEqual(cursor.reserve(), 0)
+
+        cursor.set_key(self.gen_key(95))
+        if (self.upper_bounds):
+            self.assertRaisesHavingMessage(
+                wiredtiger.WiredTigerError, lambda: cursor.reserve(), '/item not found/')
+        else:
+            self.assertEqual(cursor.reserve(), 0)
+        self.session.commit_transaction()
+
+        # Test bound API: test modifies on an existing key outside of bounds.
+        if (not self.use_colgroup):
+            self.session.begin_transaction()
+            cursor.set_key(self.gen_key(10))
+            mods = [wiredtiger.Modify("2", 0, 1)]
+            if (self.lower_bounds):
+                self.assertEqual(cursor.modify(mods), wiredtiger.WT_NOTFOUND)
+            else:
+                self.assertEqual(cursor.modify(mods), 0)
+
+            cursor.set_key(self.gen_key(95))
+            mods = [wiredtiger.Modify("2", 0, 1)]
+            if (self.upper_bounds):
+                self.assertEqual(cursor.modify(mods), wiredtiger.WT_NOTFOUND)
+            else:
+                self.assertEqual(cursor.modify(mods), 0)
+            self.session.commit_transaction()
+        
+        # Test bound API: test removing on an existing key outside of bounds.
```

Comment:
It could be worth adding boundary tests here. 
e.g. if our upper bound is key 50 what happens if we try to read/write to key 50? If this is allowed do we disallow 51?